### PR TITLE
#3 Share recognized text

### LINKED ISSUE #3: Share recognized text
As a user I want to be able to share recognized text with my friends e.g. through SMS.

### PR DESCRIPTION
**Pull request by daniellojewski**
Closes #3 
* Added icon buttons in camera preview to share or save recognized text.
* Implemented text edition before sharing or saving.
* In both cases text is stored in persistent storage to enable providing history to the user (see. #4 ). 
* Share action results in usage of OS text share option

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,4 +45,15 @@ Since ML Kit Vision is a Firebase library App has corresponding App in Firebase
 For security reasons `google-services.json` file is excluded from this repo.
 If you need access to that file get in touch via `Issues` tab.  
 Or create new project using instruction placed [here](https://codelabs.developers.google.com/codelabs/flutter-firebase/#6)  
-File `google-services.json` should be placed in Android `app` module.
```
Comment (bkraszewski) [inline README.md:44]: we'd need to document ios guide as well, info about placing google-services.plist in ios folder

File: lib/text_recognition/save/widget.dart [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+class _SaveWidgetState extends State<SaveWidget> {
+  SaveBloc _bloc;
+  final TextEditingController _textController = TextEditingController();
+  final FocusNode _textFocus = FocusNode();
+
+  @override
+  Widget build(BuildContext context) {
+    _bloc = SaveBloc(repository: RepositoryProvider.of(context));
```
Comment (bkraszewski) [inline lib/text_recognition/save/widget.dart:30]: consider using BlocProvider to manage blocs  https://github.com/felangel/bloc/blob/e818f90e5b08688f3ddfe9ea297d10be8a9249e3/docs/_snippets/flutter_firestore_todos_tutorial/main.dart.md


NVM I see you use it below :D

File: lib/text_recognition/save/widget.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        ),
+      ],
+    );
+  }
+
+  @override
+  void dispose() {
+    _textFocus.dispose();
+    _textController.dispose();
+    _bloc.close();
```
Comment (bkraszewski) [inline lib/text_recognition/save/widget.dart]: don't close bloc by yourself, BlocProvider is handling it already on removing widget from tree

File: lib/text_recognition/share/widget.dart [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        ),
+      ],
+    );
+  }
+
+  @override
+  void dispose() {
+    _textFocus.dispose();
+    _textController.dispose();
+    _bloc.close();
```
Comment (bkraszewski) [inline lib/text_recognition/share/widget.dart]: same, no need to close bloc

**Review by bkraszewski — COMMENTED** [review]
Reviewed, added few minor notes

File: README.md (hunk 1 above)
Comment (daniellojewski) [reply README.md:44]: Yeah, I know but the thing is that I don't have macOS so I'm omitting iOS configuration since the beginning

Comment (daniellojewski) [reply README.md:44]: So I'd expect that iOS will have troubles compiling but don't have option to check this. On the other hand I'll rise an issue to represent iOS configuration/setup for all dependencies

Comment (daniellojewski) [reply README.md:44]: #9 for iOS config work
#10 for readme update with disclaimer that iOS needs attention

File: pubspec.yaml [hunk 5]
```diff
@@ -35,9 +35,20 @@ dependencies:
...
   firebase_ml_vision: ^0.9.4
 
+  share: ^0.6.4+3
+
+  moor: ^3.2.0
+  moor_ffi: ^0.7.0
+  path_provider:
+  path:
+
+  fluttertoast: ^6.0.1
```
Comment (daniellojewski) [inline pubspec.yaml]: Redundant

**Pull request merged**